cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/TimePartitionedFileSetArguments.java

```diff
@@ -54,6 +60,41 @@ public static Long getOutputPartitionTime(Map<String, String> arguments) {
   }
 
   /**
+   * This is the file extension for each partition when using TimePartitionedFileSet as an OutputFormatProvider.
+   * It's used to generate the end of the output file path for each partition.
+   * @param pathFormat The format for the path; for example: 'yyyy-MM-dd/HH-mm,America/Los_Angeles',
+   *                   which will create a file path ending in the format of 2015-01-01/20-42,
+   *                   with the time of the partition being the time in the timezone of Los Angeles (PST or PDT).
+   *                   The pathFormat will be the format provided to
+   *                   {@link java.text.SimpleDateFormat}. If left blank, then the partitions will be of the form
+   *                   2015-01-01/20-42.142017372000, with the time being the time UTC.
+   *                   Note that each partition must have a unique file path or a runtime exception will be thrown.
+   */
+  public static void setOutputPathFormat(Map<String, String> arguments, String pathFormat) {
+    long curTime = System.currentTimeMillis();
+    try {
+      SimpleDateFormat format = new SimpleDateFormat(pathFormat.split(",")[0]);
+      if (pathFormat.split(",").length > 1) {
+        format.setTimeZone(TimeZone.getTimeZone(pathFormat.split(",")[1]));
```

Comment:
A comma is a valid character to have in a file path, so this parsing may break. I think its better to have time zone as a separate argument. Could have one method with just the path format which defaults to utc, then another that also takes in a time zone.